database/scrapers/faq_script.ts

```diff
@@ -0,0 +1,190 @@
+import { DateTime } from "luxon";
+import { Readable } from "node:stream";
+
+import logger from "@adonisjs/core/services/logger";
+
+import GuideArticle from "#models/guide_article";
+import GuideQuestion from "#models/guide_question";
+import FilesService from "#services/files_service";
+
+interface GuideArticlesOld {
+  data: {
+    id: number;
+    name: string;
+    cover: string;
+    short_description: string;
+    description: string | null;
+    order: number;
+    questions: number[];
+  }[];
+}
+
+interface GuideQuestionsOld {
+  data: {
+    id: number;
+    status: string;
+    date_created: string;
+    date_updated: string;
+    question: string;
+    answer: string;
+    type: number;
+  }[];
+}
+
+interface PivotTable {
+  data: {
+    id: number;
+    FAQ_Types_id: number | null;
+    FAQ_id: number;
+    sort: number | null;
+  }[];
+}
+
+interface ImageMetadata {
+  data: {
+    type: string;
+  };
+}
+
+async function uploadImage(imageUrl: string) {
+  const filesService = new FilesService();
+
+  const [fetchedImage, fetchedImageMetadata] = await Promise.all([
+    fetch(`https://admin.topwr.solvro.pl/assets/${imageUrl}`),
+    fetch(`https://admin.topwr.solvro.pl/files/${imageUrl}`),
+  ]);
+
+  if (!fetchedImage.ok) {
+    throw new Error(
+      `Failed to fetch the image. HTTP status: ${fetchedImage.status}`,
+    );
+  }
+
+  if (!fetchedImageMetadata.ok) {
+    throw new Error(
+      `Failed to fetch the image metadata. HTTP status: ${fetchedImageMetadata.status}`,
+    );
+  }
+
+  const imageMetadata = (await fetchedImageMetadata.json()) as ImageMetadata;
+  const extension = imageMetadata.data.type.split("/")[1];
+
+  const stream = Readable.fromWeb(fetchedImage.body as ReadableStream);
+
+  try {
+    return await filesService.uploadStream(stream, extension);
+  } catch (error) {
+    throw new Error(`Failed to upload image. ${error}`);
+  }
+}
```

Comment:
"Failed to upload image" adds no extra context.
Any error from `FileService.upload*()` will already have the top message say "Couldn't upload the file".
Either include what image you were unable to migrate, or don't catch the error.
If you decide to include more context, then please set the original error as the cause instead of putting only it's message in the error, as currently you're losing basically all error information. (see the [MDN docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/Error) for more info/examples on how to do this)